addons/event_sale/models/sale_order.py

```diff
@@ -14,8 +14,8 @@ def write(self, vals):
         only. """
         result = super(SaleOrder, self).write(vals)
         if vals.get('partner_id'):
-            registrations_toupdate = self.env['event.registration'].search([('sale_order_id', 'in', self.ids)])
-            registrations_toupdate.write({'partner_id': vals['partner_id']})
+            registrations_toupdate = self.sudo().env['event.registration'].search([('sale_order_id', 'in', self.ids)])
```

Comment:
Seems legit, could refer to your doppelganger's commit https://github.com/odoo/odoo/commit/19878d011f8abdc737b7b702c4c79632670d4b78 that trigerred https://github.com/odoo/odoo/commit/fb17edd0b0d5c956f6991feabfa704566fa0a11b used in '_update_registrations'.

Side note: isn't it valid for 15 ? Didn't check the code :)

Reply:
@tde-banana-odoo  I've added a PR for 15.0
 https://github.com/odoo/odoo/pull/151506